roster2/src/java/org/sakaiproject/roster/impl/SakaiProxyImpl.java

```diff
@@ -947,8 +934,8 @@ public RosterSite getRosterSite(String siteId) {
 			rosterSite.setSiteGroups(siteGroups);
 		}
 
-        Map<String, Integer> roleCounts = new HashMap<String, Integer>();
-		List<String> userRoles = new ArrayList<String>();
+        Map<String, Integer> roleCounts = new HashMap<>();
+		List<String> userRoles = new ArrayList<>();
```

Comment:
Set or List?

Reply:
@ottenhoff The lists aren't the problem. It was a layered click handler inserting the first page multiple times. Could use Sets, but there's not been an issue.